src/components/VersionChooser/index.tsx

```diff
@@ -36,6 +40,22 @@ type Props = PublicProps &
   PropsFromState &
   RouterProps;
 
+export const higherVersionsThan = (versionId: string) => {
+  if (!versionId) {
```

Comment:
I get why you need this but it indicates a flawed design. It's fine as is but the problem can be fixed in two ways. These are not change requests, just patterns to consider for future scenarios.

1. We could give storybook a utility that creates fake router locations to simulate data normally extracted from the URL. We probably have some of this already in the test helper.
2. We could make parameters such as `baseVersionId` required (which they are) and wrap the component in a helper like this:

```tsx
const VersionChooserFromRouter = (props) => {
  const { match, ...otherProps } = props;
  const { baseVersionId, headVersionId } = match.params;
  return (
    <VersionChooser
      baseVersionId={baseVersionId}
      headVersionId={headVersionId}
      {...otherProps}
    />
  );
};

export default withRouter(VersionChooserFromRouter);
```
The storybook code would then have an easier time rendering `<VersionChooser />`

Reply:
> 1. We could give storybook a utility that creates fake router locations to simulate data normally extracted from the URL. We probably have some of this already in the test helper.

That's unfortunately hard. I tried and there is nothing we can do to make everyone (React Router, TS, etc) happy it seems. It works in the test suite because we're not using a `Router` component. Wrapping the component into a `Route` would do the trick.

> 2. We could make parameters such as baseVersionId required (which they are) and wrap the component in a helper like this

Even more code to write \o/ That's probably the best solutions (of the 2), but writing such a component _just_ because Storybook isn't React Router friendly is... not fun.

I think we should make Storybook more React Router friendly, and that would mean adopting something like https://www.npmjs.com/package/storybook-react-router.

Reply:
I guess it also means it's time to fix this issue between Storybook and React Router, but maybe in a follow-up?

Reply:
> Even more code to write \o/ 

Ha, well, it was just an example. It could probably be made generic for all components like this:

```tsx
<RouterProvider>
  {(match) => (
      <VersionChooser
        baseVersionId={match.params.baseVersionId}
        headVersionId={match.params.headVersionId}
        {...otherProps}
      />
  )}
</RouterProvider>
```

> ...but writing such a component just because Storybook isn't React Router friendly is... not fun.

It highlights that the component is coupled to router in an inflexible way but, yeah, wrapping it in another component adds more complexity.

In any case, I'm not sure if we should file a follow-up or not. It's something to consider for the next component that needs a storybook *and* URL parameters.

Reply:
This makes me think that we could maybe export the component without the router HOC before the `export default`.

Reply:
I took this approach by creating a `ConnectedVersionChooser`, which accepts a `match` prop.